src/controllers/expense.controller.js

```diff
@@ -0,0 +1,105 @@
+const expenseService = require('../services/expense.service.js');
+const { checkIsReqBodyValid } = require('../helpers/checkIsReqBodyValid.js');
+
+const get = (req, res) => {
+  if (Object.entries(req.body).length === 0) {
+    res.send(expenseService.getAll());
+  }
+
+  const { ...params } = req.body;
+
+  const listOfExpectedParams = [
+    { key: 'userId', type: 'string' },
+    { key: 'categories', type: 'Array string' },
+    { key: 'from', type: 'string($date-time)' },
+    { key: 'to', type: 'string($date-time)' },
+  ];
+  const isReqBodyValid = checkIsReqBodyValid(params, listOfExpectedParams);
+
+  if (!isReqBodyValid) {
```

Comment:
Consider moving validations to the middleware.

Reply:
thanks for comment!
i improved the code